tests/test_jsvalidation.py

```diff
@@ -0,0 +1,597 @@
+            },
+            "items": {
+                "type": "array",
+                "items": {
+                    "type": "string"
+                }
+            },
+            "dependencies": {
+                "properties": {
+                    "dependencies": {
+                        "type": "integer"
+                    }
+                },
+                "dependencies": {
+                    "dependencies": {
+                        "type": "object",
+                        "properties": {
+                            "dependencies": {
+                                "type": "string"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_no_conflict_between_schemas_pass():
+    """Test that having two correct schemas and no conflict between them
+    passes."""
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+    obj.validate(v2, 'second')
+
+
+def test_conflict_between_schemas():
+    """Test that having correct schemas and conflict between them fails."""
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    v3 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"},
+            "field_A": {"type": "number"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+    obj.validate(v2, 'second')
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v3, 'third')
+
+
+def test_root_object_explicit_pass():
+    """Test that having explicit type 'object' of the root passes."""
+    v1 = {
+        "type": "object",
+        "properties": {
+            "type": "string"
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_root_object_implicit_pass():
+    """Test that having implicit type 'object' of the root passes."""
+    v1 = {
+        "properties": {
+            "type": "string"
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_root_not_object():
+    """Test that not having type 'object' of the root fails."""
+    v1 = {
+        "fieldA": {
+            "type": "string"
+        }
+    }
```

Comment:
This JSON Schema does not say validate anything. Could you replace it with the following one please?
```
v1 = {
    "type": "string"
}
```
This one has a better reason to fail as it says explicitly that the root type is "string" instead of "object".